src/httpserver/deferred_response.hpp

```diff
@@ -35,26 +35,33 @@ namespace details
     ssize_t cb(void*, uint64_t, char*, size_t);
 };
 
-typedef ssize_t(*cycle_callback_ptr)(char*, size_t);
+typedef ssize_t(*cycle_callback_ptr)(void*, char*, size_t);
```

Comment:
When I was thinking about this feature, I tried to figure how we could avoid the "void*" for the type.

To be honest this is probably one of those cases where "seeing it after it is written might help".

I was thinking we could use a template type to define the first parameter and a "shared_ptr" to pass it through. This would save us from having to use the "void*" and, ideally, from having the cleanup function entirely.

To be honest, I would convert the currently existing "char*" into a shared_ptr if we are perturbing the interface.